### ISSUE TITLE
Cases are too easy to delete

### ISSUE THREAD

**Issue by ecapuano** [TheHive-Project/TheHive#1185]
### Request Type
Feature Request

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| TheHive version / git hash   | 3.4.0-1

### Problem Description
Running TheHive within our Network Defense Range, we expose the tool to many new analysts who are unfamiliar with it... A trend I have noticed across hundreds of new users, is the accidental deletion of a case when a task deletion was intended. Long ago I made this same mistake myself during a real world incident which sucked. 

My feedback is simply to say that it is too easy to delete a case as it's not always immediately obvious the scope that is covered by a case-wide "Remove" button that is visible while looking at a task. 

![image](https://user-images.githubusercontent.com/4219694/71020821-099dae00-20cb-11ea-939b-680350c27ec0.png)

And yes, before it's mentioned in the thread, I am aware of the "Are you sure" dialogue... 
![image](https://user-images.githubusercontent.com/4219694/71021066-8a5caa00-20cb-11ea-9bbb-677a93d2b390.png)

But experience has proven that time and time again, analysts are mistaking this "Remove" function to be limited to the scope of the tab they are viewing (tasks, etc).

### Steps to Reproduce
N/A

### Possible Solutions
Perhaps move the "Remove" button from where it currently is to the Case Details page so that you can only access it when deliberately examining the case as a whole. 

Another option, change the text to specify what is being removed... ie. "Remove" -> "Remove Case"... but I still think the first option might be better...

Discuss?

### Complementary information
Covered above.

**Comment by veeral-patel** [comment]
@ecapuano Hey Eric. Here are some ideas:

1. Soft delete cases. This way, if a user deletes a case, an admin would be able to restore it within, say 30 days, of deletion.

2. Require user to enter some text to delete a case, like deleting a repo on GitHub:

<img width="493" alt="Screen Shot 2019-12-17 at 2 00 55 PM" src="https://user-images.githubusercontent.com/12554095/71037974-a9603980-20d5-11ea-8564-5e75a8313711.png">

3. Restrict who can delete a case. Maybe only an admin or the case's creator can delete a case?

**Comment by nadouani** [comment]
Hello @ecapuano the issue is probably somewhere else, since this feature is available to users with `admin` role. 

Are all your users administrators? Users with just `write` role can do almost everything except removing data.

**Comment by ecapuano** [comment]
Yes perhaps all of the users are admin because this is a training environment, but I also admit that I have done this myself inside of my production instance at 4am after a 12 hour IR....

I honestly feel some UI improvements could me made to reduce the risk, as RBAC still permits admins to make this mistake. 

I like a lot of @veeral-patel's suggestions, specifically 1, 2 & 6.

**Comment by joaociocca** [comment]
Dunno how this is coming along, but I just had an analyst ask me about archiving cases and found this discussion.
Personally, I don't see a reason for it - but I'm not a daily user, they are. Any news on this?

**Comment by ecapuano** [comment]
This is still a recurring problem... I expose TheHive to hundreds of new users per year... and I am telling you guys, this happens far more often than it should. It's happened twice today to a couple new teams.

**Comment by ecapuano** [comment]
Having to include warnings as a regular part of our trainings 
![image](https://user-images.githubusercontent.com/4219694/86946355-dc205c80-c10f-11ea-8406-540ad82cf908.png)

**Comment by ecapuano** [comment]
This happened to one of our student teams again today :(

**Comment by SophosRapidResponse** [comment]
We have done this as well (twice) it's incredibly annoying and a huge waste of time to manually re-create a case. Can we please have a fix for this in v4. Simplest idea in my mind is it just archives the case and you can get it back.
